aziotd/src/main.rs

```diff
@@ -147,7 +147,7 @@ async fn run<TMain, TConfig, TFuture, TServer>(
     config_directory_default: &str,
 ) -> Result<(), Error>
 where
-    TMain: FnOnce(TConfig) -> TFuture,
+    TMain: FnOnce(TConfig, std::path::PathBuf, std::path::PathBuf) -> TFuture,
```

Comment:
`TMain: FnOnce(TConfig, &'_ std::path::Path, &'_ std::path::Path) -> TFuture`

Let the `TMain` convert them to owned if it needs to.

(If that doesn't compile because of "'_ not allowed", `TMain: for<'a> FnOnce(TConfig, &'a std::path::Path, &'a std::path::Path) -> TFuture` should.)

Reply:
On a related note, I'm not a huge fan of using common types like `Path` or `PathBuf` as part of a function pointer type, since without parameter names, it makes it hard to tell what types are supposed to represent.

Consider bundling these two paths into a struct with a more descriptive name, or if that's too high-effort, you can just add a doc-comment describing what the parameters represent.

Reply:
Could also replace `<TMain: FnOnce(...)>(main: TMain)` with `main: fn(config: TConfig, config_path: &Path, config_dir: &Path)`, since `fn` definitions allow parameter names. It's less general but all our callers use a `fn` anyway.

Reply:
Thanks for the patch @arsing. Fixed.